lib/ansible/module_utils/basic.py

```diff
@@ -1658,18 +1645,21 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                             os.close(tmp_dest_fd)
                             # leaves tmp file behind when sudo and not root
                             try:
-                                shutil.move(b_src, b_tmp_dest_name)
+                                shutil.move(b_src, b_tmp_dest_name, copy_function=shutil.copy if keep_dest_attrs else shutil.copy2)
                             except OSError:
                                 # cleanup will happen by 'rm' of tmpdir
                                 # copy2 will preserve some metadata
-                                shutil.copy2(b_src, b_tmp_dest_name)
+                                if keep_dest_attrs:
+                                    shutil.copy(b_src, b_tmp_dest_name)
+                                else:
+                                    shutil.copy2(b_src, b_tmp_dest_name)
```

Comment:
Isn't this reverted (same above in the `shutil.move` invocation)? `shutil.copy2` preserves metadata as per above comment and the documentation. So this doesn't align with `keep_dest_attrs`?

Reply:
`keep_dest_attrs` uses `shutil.copy` to avoid overwriting the dest using tmpfile src attributes, so I think it's correct now (I definitely had it backwards while working on this though). Changing either would cause the copy tests you added in https://github.com/ansible/ansible/pull/51868 to fail.

Reply:
Thank you for the review! Naming is hard, any bikeshedding welcome.